pkg/platform/api/graphql/model/buildplanner/buildScript.go

```diff
@@ -35,57 +38,58 @@ func (o Operation) String() string {
 // TODO: We will likely need some sort of parser or other solution for the build graph
 func NewBuildScript() *BuildScript {
 	return &BuildScript{
-		Let: LetStatement{
-			Runtime: Runtime{
-				SolveLegacy: SolveLegacy{
-					Requirements: []Requirement{},
+		Let: &LetStatement{
+			Runtime: &Runtime{
+				SolveLegacy: &SolveLegacy{
+					Requirements: []*Requirement{},
 				},
 			},
 		},
 	}
 }
 
+// TODO: We may want to move this out of the model package
 type BuildScript struct {
-	Let LetStatement `json:"let"`
+	Let *LetStatement `json:"let" yaml:"let"`
 }
 
 type LetStatement struct {
-	In      string  `json:"in"`
-	Runtime Runtime `json:"runtime"`
+	In      string   `json:"in" yaml:"in"`
+	Runtime *Runtime `json:"runtime" yaml:"runtime"`
 }
 
 type Runtime struct {
-	Solve       Solve       `json:"solve,omitempty"`
-	SolveLegacy SolveLegacy `json:"solve_legacy,omitempty"`
+	// Solve       *Solve       `json:"solve,omitempty" yaml:"solve,omitempty"`
+	SolveLegacy *SolveLegacy `json:"solve_legacy,omitempty" yaml:"solve_legacy,omitempty"`
 }
 
 type Solve struct {
-	BuildFlags    []string      `json:"build_flags,omitempty"`
-	CamelFlags    []string      `json:"camel_flags,omitempty"`
-	Platforms     []string      `json:"platforms"`
-	SolverVersion string        `json:"solver_version,omitempty"`
-	AtTime        string        `json:"at_time"`
-	Requirements  []Requirement `json:"requirements"`
+	BuildFlags    []string       `json:"build_flags,omitempty" yaml:"build_flags,omitempty"`
+	CamelFlags    []string       `json:"camel_flags,omitempty" yaml:"camel_flags,omitempty"`
```

Comment:
These should still be included even if empty. See the build expression [here](https://docs.google.com/document/d/1WlnbG20q0Lq3tTKsbpZB5JsK_u9qcg1x4iG9A1AsMUo/edit#heading=h.skrjork327yz) for reference